lib/PuppeteerSharp/Puppeteer.cs

```diff
@@ -38,15 +39,15 @@ public class Puppeteer
         /// for a description of the differences between Chromium and Chrome.
         /// <a href="https://chromium.googlesource.com/chromium/src/+/lkcr/docs/chromium_browser_vs_google_chrome.md">This article</a> describes some differences for Linux users.
         /// </remarks>
-        public static Task<Browser> LaunchAsync(LaunchOptions options, int chromiumRevision)
-            => new Launcher().LaunchAsync(options, chromiumRevision);
+        public static Task<Browser> LaunchAsync(LaunchOptions options, int chromiumRevision, params ILoggerProvider[] providers) 
+            => new Launcher().LaunchAsync(options, chromiumRevision, providers);
 
         /// <summary>
         /// Attaches Puppeteer to an existing Chromium instance. The browser will be closed when the Browser is disposed.
         /// </summary>
         /// <param name="options">Options for connecting.</param>
         /// <returns>A connected browser.</returns>
-        public static Task<Browser> ConnectAsync(ConnectOptions options)
-            => new Launcher().ConnectAsync(options);
+        public static Task<Browser> ConnectAsync(ConnectOptions options, params ILoggerProvider[] providers)
```

Comment:
Why would we expect a ILoggerProvider[] instead of a ILoggerFactory?

Reply:
Another thing I'm thinking is getting rid of these static methods to be more DI-friendly. Our launcher could look like this:
```
public class Launcher
{
    public Launcher() : this(someDefault)
    {
    }
    public Launcher(ILoggerFactory factory)
    {
    }
   
    public ConnectAsync(ConnectionOptions options)
    {
    }
}
```

Using this pattern a user would be able to inject a logger if she's using a DI. And, if not, you could assign that in the constructor.

Reply:
I case you didn't read this @Meir017  ☝️